### ISSUE TITLE
Awesome!!

### ISSUE THREAD

**Issue by uBadRequest** [fokx/socks2http#1]
I was looking for an implementation like this in rust! Awesome!

Could you add the ability to round robin or any load balancing method that distributes the connections across 100 or so upstream socks proxy servers and support authentication?

**Comment by fokx** [comment]
Glad that you find the program useful.
As for load balancing, I think we can pass traffic to different SOCKS5 proxy server in `async_socks5::connect(&mut outbound, (req_host, req_port), None).await.unwrap();`. BTW, I can't figure out the meaning/scenario of doing that because I think SOCKS5 is better than HTTP proxy in that it can also pass DNS query to proxy server. I started the project as I found some functionality/performance issues when using Privoxy.
 As for authentication funcitonality, I think I would add one after I solved the problem of bare IPv6 address request to the proxy server.
